bolero/representation/promp_behavior.py

```diff
@@ -0,0 +1,497 @@
+            self.t += self.dt
+
+    def can_step(self):
+        """Returns if step() can be called again.
+
+        Note that calling step() after this function returns False will not
+        result in an error. The velocity and acceleration will be set to 0
+        and we hold the last position instead.
+
+        Returns
+        -------
+        can_step : bool
+            Can we call step() again?
+        """
+        return self.t <= self.execution_time
+
+    def get_n_params(self):
+        """Get number of weights.
+
+        Returns
+        -------
+        n_params : int
+            Number of promp weights
+        """
+        random_variables = len(self.data.mean_)
+        if self.learnCovariance:            
+            correlation_coefficients = (len(self.data.covariance_)-random_variables)/2
+            return 2*random_variables + correlation_coefficients
+        else:
+            return random_variables
+
+    def get_params(self):
+        """Get current weights. 
+
+        Returns
+        -------
+        params : array-like, shape = (n_params,)
+            Current weights
+        """
+        random_variables = self.data.mean_
+        if self.learnCovariance:
+            cov = np.array(self.data.covariance_).reshape(len(random_variables),len(random_variables))
+            Dinv =np.linalg.inv(np.sqrt(np.diag(np.diag(cov))))
+            cor = Dinv.dot(cov.dot(Dinv))
+
+            return random_variables + np.sqrt(np.diag(cov)).tolist() + np.arctanh(cor[np.tril_indices(len(random_variables),k=-1)]).tolist()
+        else:
+            return random_variables
+        
+
+    def set_params(self, params):
+        """Set new weights.
+
+        Parameters
+        ----------
+        params : array-like, shape = (n_params,)
+            New weights
+        """
+        old_cov = self.data.covariance_[:]
+        old_means = self.data.mean_[:]
+        self.data.mean_ = params[:len(self.data.mean_)]
+        if self.learnCovariance:
+            D = np.diag(np.square(np.array(params[len(self.data.mean_):2*len(self.data.mean_)])))
+            cor = np.identity(len(self.data.mean_))
+            cor[np.tril_indices(len(self.data.mean_),k=-1)] = np.tanh(params[2*len(self.data.mean_):])
+            cor[np.triu_indices(len(self.data.mean_),k=1)] = np.tanh(params[2*len(self.data.mean_):][::-1])
+            self.data.covariance_ = D.dot(cor.dot(D)).flatten().tolist()
+        self.y, self.yd, self.covars = self.trajectory()
+        
+
+    def reset(self):
+        """Reset promp."""
+        if self.x0 is None:
+            self.last_y = np.zeros(self.n_task_dims)
+        else:
+            self.last_y = np.copy(self.x0)
+        self.last_yd = np.copy(self.x0d)
+        self.last_ydd = np.copy(self.x0dd)
+
+        #self.y = np.empty(self.n_task_dims)
+        #self.yd = np.empty(self.n_task_dims)
+
+
+        self.last_t = 0.0
+        self.t = 0.0
+
+    def imitate(self, X, Xd=None, Xdd=None, alpha=0.0,
```

Comment:
`alpha` is not used. However, @AlexanderFabisch maybe keep it for consistency with `dmp_behavior.py`?